ldes-server-retention/src/main/java/be/vlaanderen/informatievlaanderen/ldes/server/retention/repositories/MemberPropertiesRepository.java

```diff
@@ -18,11 +23,21 @@ public interface MemberPropertiesRepository {
 
 	List<MemberProperties> getMemberPropertiesOfVersionAndView(String versionOf, String viewName);
 
-	Stream<MemberProperties> getMemberPropertiesWithViewReference(String viewName);
+	Stream<MemberProperties> getMemberPropertiesWithViewReference(ViewName viewName);
 
 	void removeViewReference(String id, String viewName);
 
 	void removeMemberPropertiesOfCollection(String collectionName);
 
 	void deleteById(String id);
+
+	/**
+	 * Finds all
+	 *
+	 * @param viewName
+	 * @param policy
+	 */
+	Stream<MemberProperties> findExpiredMemberProperties(ViewName viewName, TimeBasedRetentionPolicy policy);
+	Stream<MemberProperties> findExpiredMemberProperties(ViewName viewName, VersionBasedRetentionPolicy policy);
+	Stream<MemberProperties> findExpiredMemberProperties(ViewName viewName, TimeAndVersionBasedRetentionPolicy policy);
```

Comment:
So based on this, every new Retention policy would require a new method in here?